src/sentry/workflow_engine/detectors/error.py

```diff
@@ -0,0 +1,173 @@
+
+    @classmethod
+    def apply(cls, data: dict[str, Any]):
+        data_dict = cls.data_dict(**data)
+        cutoff = data_dict.cutoff
+
+        project = Project.objects.get_from_cache(id=data_dict.project_id)
+        age = project.get_option("sentry:resolve_age", None)
+        if not age:
+            return
+
+        project.update_option("sentry:_last_auto_resolve", int(time()))
+
+        if cutoff:
+            cutoff = datetime.fromtimestamp(cutoff, timezone.utc)
+        else:
+            cutoff = django_timezone.now() - timedelta(hours=int(age))
+
+        filter_conditions = {
+            "project": project,
+            "last_seen__lte": cutoff,
+            "status": GroupStatus.UNRESOLVED,
+        }
+
+        enabled_auto_resolve_types = [
+            group_type.type_id
+            for group_type in grouptype.registry.all()
+            if group_type.enable_auto_resolve
+        ]
+        filter_conditions["type__in"] = enabled_auto_resolve_types
+
+        chunk_size = data_dict.chunk_size
+        queryset = list(Group.objects.filter(**filter_conditions)[:chunk_size])
+
+        for group in queryset:
+            happened = Group.objects.filter(
+                id=group.id,
+                status=GroupStatus.UNRESOLVED,
+                last_seen__lte=cutoff,
+            ).update(
+                status=GroupStatus.RESOLVED,
+                resolved_at=django_timezone.now(),
+                substatus=None,
+            )
+            remove_group_from_inbox(group, action=GroupInboxRemoveAction.RESOLVED)
+
+            if happened:
+                Activity.objects.create(
+                    group=group,
+                    project=project,
+                    type=ActivityType.SET_RESOLVED_BY_AGE.value,
+                    data={"age": age},
+                )
+                record_group_history(group, GroupHistoryStatus.AUTO_RESOLVED)
+
+                kick_off_status_syncs.apply_async(
+                    kwargs={"project_id": group.project_id, "group_id": group.id}
+                )
+
+                analytics.record(
+                    "issue.auto_resolved",
+                    project_id=project.id,
+                    organization_id=project.organization_id,
+                    group_id=group.id,
+                    issue_type=group.issue_type.slug,
+                    issue_category=group.issue_category.name.lower(),
+                )
+                # auto-resolve is a kind of resolve and this signal makes
+                # sure all things that need to happen after resolve are triggered
+                # examples are analytics and webhooks
+                issue_resolved.send_robust(
+                    organization_id=project.organization_id,
+                    user=None,
+                    group=group,
+                    project=project,
+                    resolution_type="autoresolve",
+                    sender="auto_resolve_issues",
+                )
```

Comment:
Is there a helper method anywhere that wraps this logic up? it seems like this is the same as just saying "group this issue" and we could use or refactor an existing method.

Reply:
i've moved this logic from the auto-resolve task